tests/src/regression_main.cpp

```diff
@@ -190,8 +204,11 @@ int main(int argc, char **argv) {
         patterns.negatives.emplace_back("*/*_cmdline.*");
     }
     testing::FLAGS_gtest_filter = to_string(patterns);
-    int res = RUN_ALL_TESTS();
-    if (perf_mode && res == 0)
-        std::cout << times();
-    return res;
+#ifndef GT_STENCIL_CPU_IFIRST_HPX
+    return test_main(perf_mode);
+#else
+    hpx::util::function_nonser<int(int, char**)> run_tests =
+        hpx::util::bind(test_main, perf_mode);
+    return hpx::init(run_tests, argc, argv);
```

Comment:
So for now it is not because of the special handling of the compile definition (see [here](https://github.com/GridTools/gridtools/pull/1437/files#diff-7e7fcf44da49a920861ba2fdbecb22caR13-R15)) but I agree it's ugly, it was just to show the use of `hpx::init`